google-beta/resource_project_service_identity.go

```diff
@@ -0,0 +1,83 @@
+package google
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func resourceProjectServiceIdentity() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceProjectServiceIdentityCreate,
+		Read:   resourceProjectServiceIdentityRead,
+		Delete: resourceProjectServiceIdentityDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(20 * time.Minute),
+			Update: schema.DefaultTimeout(20 * time.Minute),
+			Read:   schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(20 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"service": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"project": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceProjectServiceIdentityCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	url, err := replaceVars(d, config, "{{ServiceUsageBasePath}}projects/{{project}}/services/{{service}}:generateServiceIdentity")
```

Comment:
Just for my own context- this began as an MM-generated file but was moved to handwritten because there's no meaning to the read/update/delete actions?

Reply:
Right. I first tried to do this in MM but saw all project service usage resources were excluded: https://github.com/GoogleCloudPlatform/magic-modules/blob/master/products/serviceusage/terraform.yaml.

And this API has a few quirks as you pointed out. Given my lack of MM expertise and the importance of this resource I chose to implement it directly in the provider. If you believe this can easily be done in MM I am happy to convert it, but may need more hand holding on that.

Reply:
All good! Handwritten code generally uses client libraries and MM code generally calls APIs directly, so I was curious.